lib/chef/provisioning/aws_driver/driver.rb

```diff
@@ -168,8 +167,8 @@ def allocate_load_balancer(action_handler, lb_spec, lb_options, machine_specs)
       # Update instance list
       actual_instance_ids = Set.new(actual_elb.instances.map { |i| i.instance_id })
 
-      instances_to_add = machine_specs.select { |s| !actual_instance_ids.include?(s.location['instance_id']) }
-      instance_ids_to_remove = actual_instance_ids - machine_specs.map { |s| s.location['instance_id'] }
+      instances_to_add = machine_specs ? machine_specs.select { |s| !actual_instance_ids.include?(s.location['instance_id']) } : []
+      instance_ids_to_remove = machine_specs ? actual_instance_ids - machine_specs.map { |s| s.location['instance_id'] } : []
 
```

Comment:
I have submitted PR https://github.com/chef/chef-provisioning/pull/286 for this part of the issue, and will merge the other bit right now.